_includes/pbc/all/install-features/202304.0/install-push-notification-feature.md

```diff
@@ -0,0 +1,698 @@
+        .catch(e => {
+          // We failed to unsubscribe, this can lead to
+          // an unusual state, so  it may be best to remove
+          // the users data from your data store and
+          // inform the user that you have done so
+          console.error('Error when unsubscribing the user', e);
+          changePushButtonState('disabled');
+        });
+  }
+
+  async function getToken() {
+    const response = await fetch('http://glue-backend.de.spryker.local/token', {
+      method: "POST",
+      body: JSON.stringify({
+        grantType: "password",
+        username: "",
+        password: ""
+      }),
+    });
+
+    return response.json()
+  }
+
+  async function push_sendSubscriptionToServer(subscription) {
+    const tokenScos = await getToken();
+    const accessToken = tokenScos[0].access_token;
+    const key = subscription.getKey('p256dh');
+    const token = subscription.getKey('auth');
+
+    return fetch('http://glue-backend.de.spryker.local/push-notification-subscriptions', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Authorization': `Bearer ${accessToken}`,
+        'Content-Type': 'application/vnd.api+json'
+      },
+      body: JSON.stringify({
+        data: {
+          type: "push-notification-subscriptions",
+          attributes: {
+            providerName: "web-push-php",
+            group: {
+              name: "warehouse",
+              identifier: 1234
+            },
+            'payload': {
+              endpoint: subscription.endpoint,
+              publicKey: key ? btoa(String.fromCharCode.apply(null, new Uint8Array(key))) : null,
+              authToken: token ? btoa(String.fromCharCode.apply(null, new Uint8Array(token))) : null,
+            }
+          }
+        }
+      }),
+    }).then(() => subscription);
+  }
+});
+```
+</details>
+
+**.../push_notification_spa/serviceWorker.js**
+
+```js
+self.addEventListener('push', function (event) {
+    if (!(self.Notification && self.Notification.permission === 'granted')) {
+        return;
+    }
+
+    const sendNotification = body => {
+        const title = "Web Push example";
+
+        return self.registration.showNotification(title, {
+            body,
+        });
+    };
+
+    if (event.data) {
+        const message = event.data.text();
+        event.waitUntil(sendNotification(message));
+    }
+});
+```
+
+5. Setup credentials:
+   1. Open `.../push_notification_spa/app.js` and replace the `applicationServerKey` variable value with your VAPID public key.
+   2. In `.../push_notification_spa/app.js`, find the `getToken()` method and replace the credentials with the user that works in your system. 
+6. Run the local HTTP server with the demo app:
+```bash
+php -S localhost:8000
+```
+
+7. To enable the feature in the system, follow the steps written in the integration guide.
+8. Enable the push notification by clicking the button on the page.
```

Comment:
Is this button obvious to the user? Probably we should specify the name of the button.